### DISCUSSION TITLE
Custom Colors

### DISCUSSION THREAD

**Discussion by askariali** [saadeghi/daisyui#2227] [Q&A]
TailwindCSS has a convention to create class names like bg-primary-main for consistent styling. I'm interested in incorporating a similar convention into the daisyUI custom theme. Could you guide how to achieve this or any recommendations on implementing such a feature?

**Comment by saadeghi** [comment]
Sorry but can you explain more clearly what you need?  

What do you mean by `consistent styling`?

Do you need more color names that are specific to a daisyUI theme?

**Reply by askariali** [reply]
I need multiple instances of one color (main, light, dark). Before this, I used to write:
`
primary:{
  main: "hex, rgb, etc",
  light: "hex, rgb, etc",
  dark: "hex, rgb, etc."
}
`
To achieve this in TailwindCSS. How can I have this feature in daisyUI?

**Reply by saadeghi** [reply]
Do you need these new color values to be different on each daisyUI theme?

**Reply by askariali** [reply]
I don't think so 🤔
In most cases, It's enough to can define our colors, but these predefined color types in [MUI](https://mui.com/material-ui/customization/palette/#color-tokens) were calculated based on the primary color. It would be awesome if we could have this approach here too.

**Reply by saadeghi** [reply]
So if you don't need those colors to be different on each theme, extending the Tailwind CSS colors would be enough:  
https://tailwindcss.com/docs/customizing-colors#adding-additional-colors  

What's the problem here?

**Reply by askariali** [reply]
I guess I made a mistake. I need those color names on each project to represent different values. Think of it as I want to have a button class called `btn-primary-dark`. Now I don't have any `btn-primary` class name instead.

I can do it manually like:
```
theme: {
  colors: {
    "primaryMain": "hex, ...",
    "primaryLight": "hex, ...",
    // ...
  }
}
```

But It is unusual and uncomfortable to use `btn-primaryMain`. On the other hand, I get some errors on something like:
```
theme: {
  colors: {
    "primary-main": "hex, ...",
    "primary-light": "hex, ...",
    // ...
  }
}
```
